kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/rbac/extractor/OauthAuthorityExtractor.java

```diff
@@ -25,7 +38,52 @@ public Mono<Set<String>> extract(AccessControlService acs, Object value, Map<Str
       throw new RuntimeException();
     }
 
-    return Mono.just(Set.of(principal.getName())); // TODO #2844
+    Set<String> groupsByUsername = acs.getRoles()
+        .stream()
+        .filter(r -> r.getSubjects()
+            .stream()
+            .filter(s -> s.getProvider().equals(Provider.OAUTH))
+            .filter(s -> s.getType().equals("user"))
+            .anyMatch(s -> s.getValue().equals(principal.getName())))
+        .map(Role::getName)
+        .collect(Collectors.toSet());
+
+    Set<String> groupsByGroupField = acs.getRoles()
+        .stream()
+        .filter(role -> role.getSubjects()
+            .stream()
+            .filter(s -> s.getProvider().equals(Provider.OAUTH))
+            .filter(s -> s.getType().equals("groupsfield"))
+            .anyMatch(subject -> convertGroups(principal.getAttribute(subject.getValue())).contains(role.getName()))
```

Comment:
I have tested this code with the following configuration for Azure AD:
```
  roles:
   - name: "Admin"
      subjects:
        - provider: OAUTH
          type: user
          value: "Aleksei Zotov"
        - provider: OAUTH
          type: groupsfield
          value: "roles"
```
Everything seems to be working as my user is a member of "Admin" group. However, I'm a bit confused with the current modeling.

Generally roles may contain multiple subjects (different providers/users/groups/domains, some examples are available at https://docs.kafka-ui.provectus.io/configuration/rbac-role-based-access-control#subjects) and there is no limitation for the role to have any specific name. However, in the current implementation, the role should have the same name as one the groups assigned to a particular user. I'm not sure it is what users expect.

In my opinion (I'm not really well familiar with kafka-ui architecture/concepts), we should do something like this:
```
String groupAttributeName; // it should be read from config file for this particular provider
Set<String> groupsByGroupField = acs.getRoles()
        .stream()
        .filter(role -> role.getSubjects()
            .stream()
            .filter(s -> s.getProvider().equals(Provider.OAUTH))
            .filter(s -> s.getType().equals("group"))
            .anyMatch(subject -> convertGroups(principal.getAttribute(groupAttributeName)).contains(subject.getValue()))
```
The configuration for such implementation would look like:
```
  roles:
   - name: "WhateverYouWant"
      subjects:
        - provider: OAUTH
          type: user
          value: "Aleksei Zotov"
        - provider: OAUTH
          type: group
          value: "Admin"
```
I feel such modeling would more transparent and clear to the end users.

Please, share your thoughts.

Reply:
@azotcsit hey!

Thanks much for your comment, really appreciate it.

I had to occupy the subject's value for the field's name. And, for some reason, it didn't come up to my mind at the moment (#3700 might've clouded my judgment lol) that I can have the param at the provider's level.

I've applied some changed, can you check again with the following config?

```
auth:
  type: OAUTH2
  oauth2:
    client:
      name:
        clientId:
        clientSecret:
        // [...]
        custom-params:
          type: oauth
          // [...]
          roles-field: roles

```

And the following RBAC config:

```
      subjects:
        - provider: oauth
          type: role
          value: Admin
```

Lemme know how it goes.

Reply:
I've tested logic with the new changes, everything seems to be working as expected. Thanks for addressing the comment!